backend/app/src/main/java/ch/zhaw/studyflow/controllers/CalendarController.java

```diff
@@ -0,0 +1,328 @@
+        return authenticator.handleIfAuthenticated(request, principal -> {
+            Optional<Long> userId = principal.getClaim(CommonClaims.USER_ID).map(Long::valueOf);
+            if (userId.isPresent()) {
+                List<Calendar> calendars = calendarManager.getCalendarsByUserId(userId.get());
+                response.setResponseBody(JsonContent.writableOf(calendars))
+                        .setStatusCode(HttpStatusCode.OK);
+            } else {
+                response.setStatusCode(HttpStatusCode.BAD_REQUEST);
+            }
+            return response;
+        });
+    }
+
+    /**
+     * Endpoint for retrieving all appointments in a specific calendar within a date range.
+     *
+     * @param context the request context
+     * @return the HTTP response
+     */
+    @Route(path = "{calendarId}/appointments")
+    @Endpoint(method = HttpMethod.GET)
+    public HttpResponse getAppointments(RequestContext context) {
+        final HttpRequest request = context.getRequest();
+        HttpResponse response = context.getRequest().createResponse();
+        return authenticator.handleIfAuthenticated(request, principal -> {
+            Optional<Long> userId = principal.getClaim(CommonClaims.USER_ID).map(Long::valueOf);
+            if (userId.isPresent()) {
+                QueryParameters queryParameters = request.getQueryParameters();
+                Date from = queryParameters.getSingleValue("from")
+                        .map(Date::new)
+                        .orElse(new Date(0));
+                Date to = queryParameters.getSingleValue("to")
+                        .map(Date::new)
+                        .orElse(new Date());
+                List<Appointment> appointments = appointmentManager.readAllBy(userId.get(), from, to);
+                response.setResponseBody(JsonContent.writableOf(appointments))
+                        .setStatusCode(HttpStatusCode.OK);
+            } else {
+                response.setStatusCode(HttpStatusCode.BAD_REQUEST);
+            }
+            return response;
+        });
+    }
+
+    /**
+     * Endpoint for retrieving a specific appointment by date.
+     *
+     * @param context the request context
+     * @return the HTTP response
+     */
+    @Route(path = "{calendarId}/appointments/{appointmentId}")
+    @Endpoint(method = HttpMethod.GET)
+    public HttpResponse getAppointmentsByDate(RequestContext context) {
+        final HttpRequest request = context.getRequest();
+        HttpResponse response = request.createResponse(); // Simplified method call
+
+        return authenticator.handleIfAuthenticated(request, principal -> {
+            Optional<Long> userId = principal.getClaim(CommonClaims.USER_ID).map(Long::valueOf);
+            if (userId.isPresent()) {
+                QueryParameters queryParameters = request.getQueryParameters();
+
+                Date from = queryParameters.getSingleValue("from")
+                        .map(Date::new)
```

Comment:
I'm not sure, but I think the new java.time API is preferred over java.util.Date.